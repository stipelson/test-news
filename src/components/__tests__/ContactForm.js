import React from 'react';
import { create, act } from 'react-test-renderer';
import Row from 'emerald-ui/lib/Row';
import Col from 'emerald-ui/lib/Col';

import ContactForm from '../ContactForm';

describe('ContactForm: render', () => {
  it('renders contact form correctly', () => {
    const tree = create(<ContactForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Section has id', () => {
    const component = create(<ContactForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const section = instance.findByType('section');
    expect(section.props.id).toEqual('contact');
  });

  it('Content has class', () => {
    const component = create(<ContactForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const content = instance.findByType('div');
    expect(content.props.className).toEqual('container');
  });

  it('Rows has class, except last', () => {
    const component = create(<ContactForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const allRows = instance.findAllByType(Row);

    for (let index = 0; index < allRows.length - 1; index++) {
      expect(allRows[index].props.className).toEqual('extra-p-md-6');
    }
  });

  it('Cols has class', () => {
    const component = create(<ContactForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const allCols = instance.findAllByType(Col);

    for (let index = 0; index < allCols.length; index++) {
      expect(allCols[index].props.className).toEqual('mb-46');
    }
  });

  it('Checkbox props render', () => {
    const component = create(
      <ContactForm title="Contact Us" onValidForm={() => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const title = instance.findByType('h2');
    expect(title.children).toEqual(['Contact Us']);
  });
});

describe('ContactForm: submit', () => {
  it('Contact form submit with empty values', () => {
    const onSubmitMock = jest.fn();
    const event = {
      preventDefault: jest.fn(),
    };
    let component;
    act(() => {
      component = create(
        <ContactForm title="Contact Us" onValidForm={onSubmitMock} />
      );
    });
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;
    const form = instance.findByType('form');

    act(() => {
      form.props.onSubmit(event);
    });
    // expect(onSubmitMock).toHaveBeenLastCalledWith(1);
    expect(onSubmitMock).toHaveBeenCalledTimes(0);
  });

  it('Contact form submit with values', () => {
    const data = {
      first_name: 'Test name',
      last_name: 'Test Last name',
      email: 'test@test.com',
      phone: '8005555555',
      message: 'Test message',
      email_subscription: false,
    };
    const onSubmitMock = jest.fn();

    const event = {
      preventDefault: jest.fn(),
      target: { name: 'fname', value: 'text' },
    };
    let component;

    // First State

    act(() => {
      component = create(
        <ContactForm title="Contact Us" onValidForm={onSubmitMock} />
      );
    });
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;
    const form = instance.findByType('form');
    const fname = instance.findByProps({ name: 'fname' });
    const lname = instance.findByProps({ name: 'lname' });
    const email = instance.findByProps({ name: 'email' });
    const phone = instance.findByProps({ name: 'phone' });
    const message = instance.findByProps({ name: 'message' });
    // Manual change input values
    act(() => {
      const inputEvent = {
        preventDefault: jest.fn(),
      };
      fname.props.onChange({
        ...inputEvent,
        target: { name: 'fname', value: 'Test name', error: '' },
      });
      lname.props.onChange({
        ...inputEvent,
        target: { name: 'lname', value: 'Test Last name' },
      });
      email.props.onChange({
        ...inputEvent,
        target: { name: 'email', value: 'test@test.com' },
      });
      phone.props.onChange({
        ...inputEvent,
        target: { name: 'phone', value: '8005555555' },
      });
      message.props.onChange({
        ...inputEvent,
        target: { name: 'message', value: 'Test message' },
      });
    });

    // Submit form
    act(() => {
      form.props.onSubmit(event);
    });

    // Check if called
    // expect(onSubmitMock).toHaveBeenCalledTimes(1);
    //  Check called mock function with form data
    expect(onSubmitMock).toHaveBeenLastCalledWith(
      JSON.stringify(data, null, 2)
    );
  });
});
