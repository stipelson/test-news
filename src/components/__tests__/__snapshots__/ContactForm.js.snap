// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm renders contact form correctly 1`] = `
<section
  id="contact"
>
  <div
    className="container"
  >
    <h2
      className="mt-0 text-center"
    >
      Contact Us
    </h2>
    <div
      className="eui-panel card-container eui-panel-default"
    >
      <div
        className="eui-panel-body"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="eui-row extra-p-md-6"
          >
            <div
              className="eui-col mb-46 eui-col-xs-6"
            >
              <div
                className="eui-text-field"
              >
                <div
                  className="eui-text-field-wrapper"
                >
                  <input
                    aria-errormessage="undefinedErrorMessage"
                    aria-invalid={false}
                    aria-label="First Name"
                    disabled={false}
                    name="fname"
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    role="textbox"
                    type="text"
                    value=""
                  />
                  <div
                    className="eui-text-field-footer"
                  >
                    <span
                      className="eui-text-field-message eui-text-field-error-message"
                      id="undefinedErrorMessage"
                      role="alert"
                    >
                      
                    </span>
                    <span
                      className="eui-text-field-message eui-text-field-help-text"
                    />
                  </div>
                </div>
                <label>
                  First Name
                </label>
              </div>
            </div>
            <div
              className="eui-col mb-46 eui-col-xs-6"
            >
              <div
                className="eui-text-field"
              >
                <div
                  className="eui-text-field-wrapper"
                >
                  <input
                    aria-errormessage="undefinedErrorMessage"
                    aria-invalid={false}
                    aria-label="Last Name"
                    disabled={false}
                    name="lname"
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    role="textbox"
                    type="text"
                    value=""
                  />
                  <div
                    className="eui-text-field-footer"
                  >
                    <span
                      className="eui-text-field-message eui-text-field-error-message"
                      id="undefinedErrorMessage"
                      role="alert"
                    >
                      
                    </span>
                    <span
                      className="eui-text-field-message eui-text-field-help-text"
                    />
                  </div>
                </div>
                <label>
                  Last Name
                </label>
              </div>
            </div>
          </div>
          <div
            className="eui-row extra-p-md-6"
          >
            <div
              className="eui-col mb-46 eui-col-xs-6"
            >
              <div
                className="eui-text-field"
              >
                <div
                  className="eui-text-field-wrapper"
                >
                  <input
                    aria-errormessage="undefinedErrorMessage"
                    aria-invalid={false}
                    aria-label="Email"
                    disabled={false}
                    name="email"
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    role="textbox"
                    type="text"
                    value=""
                  />
                  <div
                    className="eui-text-field-footer"
                  >
                    <span
                      className="eui-text-field-message eui-text-field-error-message"
                      id="undefinedErrorMessage"
                      role="alert"
                    >
                      
                    </span>
                    <span
                      className="eui-text-field-message eui-text-field-help-text"
                    />
                  </div>
                </div>
                <label>
                  Email
                </label>
              </div>
            </div>
            <div
              className="eui-col mb-46 eui-col-xs-6"
            >
              <div
                className="eui-text-field"
              >
                <div
                  className="eui-text-field-wrapper"
                >
                  <input
                    aria-errormessage="undefinedErrorMessage"
                    aria-invalid={false}
                    aria-label="Phone Number"
                    disabled={false}
                    name="phone"
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    role="textbox"
                    type="text"
                    value=""
                  />
                  <div
                    className="eui-text-field-footer"
                  >
                    <span
                      className="eui-text-field-message eui-text-field-error-message"
                      id="undefinedErrorMessage"
                      role="alert"
                    >
                      
                    </span>
                    <span
                      className="eui-text-field-message eui-text-field-help-text"
                    />
                  </div>
                </div>
                <label>
                  Phone Number
                </label>
              </div>
            </div>
          </div>
          <div
            className="eui-row"
          >
            <div
              className="eui-col mb-46 eui-col-xs-12"
              role="textbox"
            >
              <div
                className="eui-text-field  "
              >
                <div
                  className="eui-text-field-wrapper"
                >
                  <textarea
                    aria-errormessage="undefinedErrorMessage"
                    aria-invalid={false}
                    aria-label="Message"
                    className="custom-textarea"
                    name="message"
                    onChange={[Function]}
                    role="textbox"
                    rows="4"
                    value=""
                  />
                  <div
                    className="eui-text-field-footer"
                  >
                    <span
                      className="eui-text-field-message eui-text-field-error-message "
                      id="undefinedErrorMessage"
                      role="alert"
                    >
                      
                    </span>
                    <span
                      className="eui-text-field-message eui-text-field-help-text"
                    />
                  </div>
                </div>
                <label>
                  Message
                </label>
              </div>
            </div>
          </div>
          <div
            className="eui-row"
          >
            <div
              className="eui-col eui-col-xs-12"
              role="textbox"
            >
              <label
                className="eui-checkbox mt-0 mb-46 eui-has-label"
              >
                <input
                  className="eui-checkbox-input"
                  name="subscription"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                />
                <div
                  className="eui-checkbox-check"
                  onClick={[Function]}
                  role="presentation"
                />
                <span
                  className="eui-checkbox-label"
                  onClick={[Function]}
                  role="presentation"
                >
                  Send me emails about breaking news and promotions.
                </span>
              </label>
            </div>
          </div>
          <div
            className="text-center submit-button"
          >
            <button
              aria-label="Subscribe to newsletter"
              className="eui-btn eui-btn-primary eui-btn-classic eui-btn-sm eui-btn-inverted btn-padding-lg"
              role="button"
              type="submit"
            >
              <div
                className="eui-btn-content"
              >
                <span
                  className="caption"
                >
                  <span>
                    Submit form
                  </span>
                </span>
              </div>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</section>
`;
